### PR TITLE
Add content-packages-validation Analyser

### PR DESCRIPTION
**Pull request by cschneider**
## Description

Activate vault validation.

## Related Issue

#134 

## How Has This Been Tested?

mvn clean install

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java [hunk 1]
```diff
@@ -61,7 +61,8 @@
         + ",bundle-nativecode"
         + ",artifact-rules"
         + ",configuration-api"
-        + ",aem-env-var";
+        + ",aem-env-var"
+        + ",content-packages-validation";
```
Comment (kwin) [inline aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java:65]: Please don’t enable by default as modern projects use filevault-content-package-m-p already for building and validation, and executing validation twice just costs performance.

File: aemanalyser-maven-plugin/pom.xml [hunk 2]
```diff
@@ -219,6 +219,21 @@ governing permissions and limitations under the License.
       <artifactId>org.apache.sling.feature.analyser</artifactId>
       <version>1.5.2</version>
     </dependency>
+    <dependency>
```
Comment (cziegeler) [inline aemanalyser-maven-plugin/pom.xml:222]: Why are these dependencies needed here? They are already part of the core module

Comment (cschneider) [reply aemanalyser-maven-plugin/pom.xml:222]: The problem is that all dependencies in feature analyser are set to provided. 
The core module only depends on org.apache.sling.feature.analyser.
How about moving those transitive dependencies to aemanalyser core?

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java (hunk 1 above)
Comment (cschneider) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java:65]: Carsten asked me to put it there. @cziegeler what do you think?

Comment (cschneider) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java:65]: Carsten asked me to add it 
@cziegeler What do you think?

Comment (cziegeler) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java:65]: The aemanalyser is a general analyser with the intend to check all aspects of an aem project. As the analyser has no idea what has and what has not been validated already, we activate all analysers by default. If a project is already validating something as part of the build, that project can disable this analyser.
If we do it the other way, we end of with many projects where the validation is not executed as noone will actually enable it manually.

File: aemanalyser-maven-plugin/pom.xml (hunk 2 above)
Comment (cziegeler) [reply aemanalyser-maven-plugin/pom.xml:222]: Wait, these dependencies are already in this pom, look some lines above

Comment (cschneider) [reply aemanalyser-maven-plugin/pom.xml:222]: You are right. I can simply remove may addition there.

**Comment by cziegeler** [comment]
Thanks @cschneider . I think the default configuration for the analyser should be added to : https://github.com/adobe/aemanalyser-maven-plugin/blob/main/aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java#L130
Instead of disabling validators in that analyser (deny list), can we set a list of enabled validators (allow list)? I don't want to get surprised by an update of the vault validators which messes around with the default set and breaks our assumptions.

File: aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java (hunk 1 above)
Comment (kwin) [reply aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyser.java:67]: Can you please make sure that in the https://github.com/adobe/aem-project-archetype validation is only performed once during Maven build (only in filevault-package-maven-plugin) as that supports m2e and incremental builds properly? That would also mean that you need to keep the configuration for both plugins in sync with regards to validation.

**Review by cziegeler — APPROVED** [review]
Thanks for the updates

**Pull request merged**